src/marten/template/tag/localize.cr

```diff
@@ -0,0 +1,99 @@
+require "./concerns/*"
+
+module Marten
+  module Template
+    module Tag
+      # The `localize` template tag.
+      #
+      # The `localize` template tag allows performing localization on values such as dates, numbers, and times
+      # within templates. It can take one mandatory argument (the value to localize) and an optional `format`
+      # keyword argument to specify a localization format.
+      #
+      # Usage examples:
+      # ```
+      # {% localize created_at %}
+      # {% localize 100000 format: "short" %}
+      # ```
+      #
+      # Optionally, the result of the localization can be assigned to a variable using `as` keyword:
+      #
+      # ```
+      # {% localize created_at format: "short" as localized_date %}
+      # ```
+      class Localize < Base
+        include CanExtractKwargs
+        include CanSplitSmartly
+
+        @assigned_to : String? = nil
+
+        def initialize(parser : Parser, source : String)
+          parts = split_smartly(source)
+
+          if parts.size < 2
+            raise Errors::InvalidSyntax.new("Malformed localize tag: at least one argument must be provided")
+          end
+
+          @value_expression = FilterExpression.new(parts[1])
+
+          if parts.size > 2 && parts[-2] == "as"
+            @assigned_to = parts[-1]
+            kwargs_parts = parts[2...-2]
+          else
+            kwargs_parts = parts[2..]
+          end
+
+          @kwargs = {} of String => FilterExpression
+          extract_kwargs(kwargs_parts.join(' ')).each do |key, value|
+            @kwargs[key] = FilterExpression.new(value)
+          end
+        end
+
+        def render(context : Context) : String
+          value = @value_expression.resolve(context).raw
+
+          if format_value = @kwargs.delete("format")
```

Comment:
Why are we deleting the `format` from the `kwargs` instance variable here? This could make successive uses of the template tag behave inconsistently!

If believe we want to perform a regular key lookup instead:

```suggestion
          if !(format_value = @kwargs["format")]?).nil?
```